controllers/storageclassclaim_migration_controller.go

```diff
@@ -0,0 +1,117 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/red-hat-storage/ocs-client-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	// for migration of storageclassclaims to storageclaims
+	storageClassClaimFinalizer = "storageclassclaim.ocs.openshift.io"
+)
+
+// StorageClassClaimReconcile migrates StorageClassClaim objects to StorageClaim
+type StorageClassClaimMigrationReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+
+	log logr.Logger
+	ctx context.Context
+}
+
+func (r *StorageClassClaimMigrationReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.StorageClassClaim{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
```

Comment:
`GenerationChangedPredicate` does not make sense here. 
Also, we want a predicate to limit only the 'create' event.

Reply:
done.